crates/bevy_render/src/pipeline/pipeline_compiler.rs

```diff
@@ -12,7 +15,44 @@ use serde::{Deserialize, Serialize};
 
 #[derive(Clone, Eq, PartialEq, Debug, Reflect)]
 #[reflect(PartialEq)]
-pub struct PipelineSpecialization {
+pub enum PipelineSpecialization {
+    Render(RenderPipelineSpecialization),
+    Compute(ComputePipelineSpecialization),
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Reflect)]
+#[reflect(PartialEq)]
+pub struct ComputePipelineSpecialization {
+    pub shader_specialization: ShaderSpecialization,
+    pub dynamic_bindings: HashSet<String>,
+}
+
+impl Default for ComputePipelineSpecialization {
+    fn default() -> Self {
+        Self {
+            shader_specialization: Default::default(),
+            dynamic_bindings: Default::default(),
```

Comment:
```suggestion
            ..Default::default()
```

Reply:
That would call this default impl recursively. This default impl should probably just be derived.